js/src/components/contact-information/phone-number-card/verify-phone-number-content.test.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+
+/**
+ * Internal dependencies
+ */
+import VerifyPhoneNumberContent from './verify-phone-number-content';
+
+describe( 'VerifyPhoneNumberContent component', () => {
+	test( 'typing Enter submits enclosing form', async () => {
+		const onVerificationStateChange = jest.fn();
+
+		render(
+			<VerifyPhoneNumberContent
+				verificationMethod="SMS"
+				onVerificationStateChange={ onVerificationStateChange }
+			/>
+		);
+
+		await act( async () => {
+			await userEvent.type(
+				screen.getAllByRole( 'textbox' )[ 0 ],
+				'{enter}'
+			);
```

Comment:
Similar to the above, it will issue the `verifyPhoneNumber` action. We could consider mocking the action.

https://github.com/woocommerce/google-listings-and-ads/blob/e06126dc1f62b7fb98e98e03228d1b849c881d8c/js/src/components/contact-information/phone-number-card/verify-phone-number-content.js#L142-L152